utils/src/filesystem_content_loader.cpp

```diff
@@ -58,20 +58,58 @@ std::optional<std::string> filesystem_content_loader::load_text(const resource_l
 list_directory_result filesystem_content_loader::list_directory_files(
     const utils::resource::resource_location& directory_loc) const
 {
-    std::filesystem::path lib_p(directory_loc.get_path());
+    std::filesystem::path path(directory_loc.get_path());
     list_directory_result result;
 
-    result.second = utils::path::list_directory_regular_files(lib_p, [&result](const std::filesystem::path& f) {
+    result.second = utils::path::list_directory_regular_files(path, [&result](const std::filesystem::path& f) {
         result.first[utils::path::filename(f).string()] =
             utils::resource::resource_location(utils::path::path_to_uri(utils::path::absolute(f).string()));
     });
 
     return result;
 }
 
+list_directory_result filesystem_content_loader::list_directory_subdirs_and_symlinks(
+    const utils::resource::resource_location& directory_loc) const
+{
+    std::filesystem::path path(directory_loc.get_path());
+    list_directory_result result;
+
+    result.second = utils::path::list_directory_subdirs_and_symlinks(path, [&result](const std::filesystem::path& p) {
+        std::error_code ec;
+        auto cp = utils::path::canonical(p, ec);
+
+        if (!ec && utils::path::is_directory(cp))
+        {
+            auto found_dir = utils::resource::resource_location(utils::path::path_to_uri(cp.string()));
+            found_dir.join(""); // Ensure that this is a directory
+            result.first[cp.string()] = std::move(found_dir);
```

Comment:
symlinks pointing to the same directory will start overwriting each other here...

Reply:
Good point. I've switched the map parameters. I'm also open with going with a different approach if you'd like